spec/query/query_string_spec.rb

```diff
@@ -0,0 +1,155 @@
+      end
+
+      context "escaping special characters" do
+        let(:condition) { '~tex*t wi[t]h "s{pec}ial" #v@lue$!^' }
+        let(:expected_term) { '(*\~tex\*t wi\[t\]h \"s\{pec\}ial\" #v@lue$s!*)\^' }
+      end
+    end
+
+    context "when condition is a Hash" do
+      let(:condition) { { some: { nested: { condition: "filter text" } } } }
+      let(:expected_term) { 'some.nested.condition:(*filter AND text*)' }
+
+      it_behaves_like "a search term"
+
+      context "with many terms" do
+        let(:condition) { { some: { nested: { condition: "filter text", part: "two" } } } }
+        let(:expected_term) { 'some.nested.condition:(*filter AND text*) AND some.nested.part:(*two*)' }
+
+        it_behaves_like "a search term"
+      end
+
+      context "value is an Array" do
+        let(:condition) { { some: { nested: { condition: ["filter", "text"] } } } }
+        let(:expected_term) { 'some.nested.condition:((*filter*) OR (*text*))' }
+
+        it_behaves_like "a search term"
+      end
+
+      context "value is a Range" do
+        let(:condition) { { some: { nested: { condition: (1..2) } } } }
+        let(:expected_term) { "some.nested.condition:[1 TO 2]" }
+
+        it_behaves_like "a search term"
+      end
+    end
+
+    context "when condition is an Array" do
+      let(:condition) { ["filter text", "some more filter text"] }
+      let(:expected_term) { '((*filter AND text*) OR (*some AND more AND filter AND text*))' }
+
+      it_behaves_like "a search term"
+    end
+
+    context "when condition is a Numeric" do
+      let(:condition) { 1 }
+      let(:expected_term) { 1 }
+
+      it_behaves_like "a search term"
+    end
+
+    context "when condition is a Range" do
+      context "that is inclusive" do
+        let(:condition) { 1..3 }
+        let(:expected_term) { "[1 TO 3]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "that is exclusive" do
+        let(:condition) { 1...3 }
+        let(:expected_term) { "[1 TO 2]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "values are integers" do
+        it_behaves_like "a search term"
+      end
+
+      context "values are strings" do
+        let(:condition) { 'a'..'z' }
+        let(:expected_term) { "[a TO z]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "values are time objects" do
+        let(:condition) do
+          time = Time.utc(2018, 12, 27, 20, 10).in_time_zone("Pacific Time (US & Canada)")
+          min_time = time - 300
+          max_time = time + 300
+          min_time..max_time
+        end
+        let(:expected_term) { "[2018-12-27T20:05:00Z TO 2018-12-27T20:15:00Z]" }
+
+        it_behaves_like "a search term"
+      end
+
+      context "values are date objects" do
+        let(:condition) do
+          date = Date.new(2019, 2, 1)
+          min_date = date - 1
+          max_date = date + 1
+          min_date..max_date
+        end
```

Comment:
Should there be a test for an exclusive date range?

Reply:
I didn't see the need for one since I already had an explicit exclusive range spec that was meant to be agnostic of the type of object.

If you feel strongly I could add one